### DISCUSSION TITLE
The stores could not be found

### DISCUSSION THREAD

**Discussion by danieljwestman** [directus/directus#8576] [Archive]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

Building a fresh Directus clone from main repo works as it should. But when starting up some routes fails.

<img width="681" alt="Skärmavbild 2021-10-05 kl  16 24 43" src="https://user-images.githubusercontent.com/20950568/136042516-73a0ffb9-bb51-4d76-b833-8a48e361fb00.png">

Some of the failing routes are `/users`, `/files` and `/collections/[collection name]`.

The error seems to be that the stores can't be found in the `shared` package.

Line 6 in /packages/shared/src/composables/use-system.ts

<img width="678" alt="Skärmavbild 2021-10-05 kl  16 30 47" src="https://user-images.githubusercontent.com/20950568/136043623-61a5d1b6-36f1-44fd-8318-d88259e59b14.png">

`stores` beeing `undefined`.

 

### To Reproduce

Download a fresh clone of the main repo.

`npm install`

`npx directus start`

(`npm run dev` is still working)

### What version of Directus are you using?

v.9.0.0-rc.95

### What version of Node.js are you using?

16

### What database are you using?

PG 13

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying Directus?

running locally / main repo clone

**Comment by rijkvanzanten** [comment]
Make sure to run `npm run build` as well after you've ran `npm install`. I believe the issue here is that your dist from `@directus/shared` is outdated

**Comment by danieljwestman** [comment]
Yes, forgot to write that.

`npm run build` is indeed done... before running

**Comment by rijkvanzanten** [comment]
Is your local bin for `directus` the local build of the api, or could it be it's an (older) global install? I tend to use `npm run dev -w directus` to run the api in dev mode, or use `npm run cli <command>` to run the CLI commands straight from source 🤔

**Comment by danieljwestman** [comment]
directus bin is a symlink to the src (api build)..  🤯

This has been working just fine before...

**Comment by rijkvanzanten** [comment]
Was that first npm install in the root from a recent version of node / npm? I know they were having some trouble with the workspaces package resolution, so that might be another culprit 🤔  https://github.com/npm/cli/issues/3637

**Comment by danieljwestman** [comment]
Hm, okay. I'm currently running node 16, I'll give it try to downgrade

**Comment by danieljwestman** [comment]
Now tried with **node 16, 14 and 12**, and **npm 7 and 6**. 

Same problem... 😣

<img width="401" alt="Skärmavbild 2021-10-05 kl  18 47 06" src="https://user-images.githubusercontent.com/20950568/136066994-e10e77d5-f77a-4eda-9227-34c61459763c.png">

**Comment by rijkvanzanten** [comment]
@nickrum any ideas?

**Comment by danieljwestman** [comment]
(This is the same problem that lately occurs when running on DigitalOcean App Platform, as mentioned in Discord)

**Comment by nickrum** [comment]
Very odd.. @danieljwestman could you check if there is a `vue` folder inside `node_modules/`, `app/node_modules/` and `packages/shared/node_modules/`?

**Comment by danieljwestman** [comment]
✅ /node_modules/vue
✅ /app/node_modules/vue
✅ /packages/shared/node_modules/vue

The app is working for some routes, but not `/users`, `/files` and `/collections/[collection name]`, and some other.

**Comment by rijkvanzanten** [comment]
@danieljwestman What's the output of `npm list vue`? Wondering if they're using a different context

**Comment by nickrum** [comment]
That's not a good sign... Vue should only be installed inside the root `node_modules/`, otherwise app and shared will use different versions and the `inject()`-call you showed above won't work.

I'm not sure what is causing this, though. When you were trying other npm versions, did you clean the repo beforehand? (Using `npx lerna clean` and deleting the root `node_modules/`)

**Comment by danieljwestman** [comment]
> @rijkvanzanten What's the output of `npm list vue`? Wondering if they're using a different context

```
├─┬ @directus/app@9.0.0-rc.95 -> ./app
│ ├─┬ @directus/shared@9.0.0-rc.95 -> ./packages/shared
│ │ └── vue@3.2.19 deduped
│ ├─┬ @tinymce/tinymce-vue@4.0.4
│ │ └── vue@3.2.19 deduped
│ ├─┬ @vue/cli-plugin-babel@4.5.13
│ │ └─┬ @vue/babel-preset-app@4.5.13
│ │   └── vue@3.2.19 deduped
│ ├─┬ pinia@2.0.0-rc.11
│ │ ├─┬ vue-demi@0.11.4
│ │ │ └── vue@3.2.19 deduped
│ │ └── vue@3.2.19 deduped
│ ├─┬ vue-i18n@9.1.8
│ │ └── vue@3.2.19 deduped
│ ├─┬ vue-router@4.0.11
│ │ └── vue@3.2.19 deduped
│ ├─┬ vue@3.2.19
│ │ └─┬ @vue/server-renderer@3.2.19
│ │   └── vue@3.2.19 deduped
│ └─┬ vuedraggable@4.1.0
│   └── vue@3.2.19 deduped
├─┬ @directus/docs@9.0.0-rc.95 -> ./docs
│ └─┬ vuepress@1.8.2
│   └─┬ @vuepress/core@1.8.2
│     ├─┬ @vue/babel-preset-app@4.5.13
│     │ └── vue@2.6.14 deduped
│     └── vue@2.6.14
├─┬ @directus/extensions-sdk@9.0.0-rc.95 -> ./packages/extensions-sdk
│ └─┬ @directus/shared@9.0.0-rc.95 -> ./packages/shared
│   └── vue@3.2.19 deduped
├─┬ @directus/shared@9.0.0-rc.95 -> ./packages/shared
│ ├─┬ vue-i18n@9.1.8
│ │ └── vue@3.2.19 deduped
│ ├─┬ vue-router@4.0.11
│ │ └── vue@3.2.19 deduped
│ └─┬ vue@3.2.19
│   └─┬ @vue/server-renderer@3.2.19
│     └── vue@3.2.19 deduped
├─┬ @tinymce/tinymce-vue@4.0.4 extraneous
│ └── vue@3.2.19 deduped
├─┬ @vue/babel-preset-app@4.5.13 extraneous
│ └── vue@3.2.19 deduped
├─┬ @vue/server-renderer@3.2.19 extraneous
│ └── vue@3.2.19 deduped
├─┬ @vuepress/core@1.8.2 extraneous
│ └── vue@2.6.14 extraneous
├─┬ create-directus-extension@9.0.0-rc.95 -> ./packages/create-directus-extension
│ └─┬ @directus/shared@9.0.0-rc.95 -> ./packages/shared
│   └── vue@3.2.19 deduped
├─┬ directus@9.0.0-rc.95 -> ./api
│ ├─┬ @directus/app@9.0.0-rc.95 -> ./app
│ │ └── vue@3.2.19 deduped
│ └─┬ @directus/shared@9.0.0-rc.95 -> ./packages/shared
│   └── vue@3.2.19 deduped
├─┬ pinia@2.0.0-rc.11 extraneous
│ ├─┬ vue-demi@0.11.4 extraneous
│ │ └── vue@3.2.19 deduped
│ └── vue@3.2.19 deduped
├─┬ vue-i18n@9.1.8 extraneous
│ └── vue@3.2.19 deduped
├─┬ vue-router@4.0.11 extraneous
│ └── vue@3.2.19 deduped
├── vue@3.2.19 extraneous
└─┬ vuedraggable@4.1.0 extraneous
  └── vue@3.2.19 deduped
```

**Comment by danieljwestman** [comment]
> @nickrum  I'm not sure what is causing this, though. When you were trying other npm versions, did you clean the repo beforehand? (Using `npx lerna clean` and deleting the root `node_modules`)

I cleaned up with `npx lerna clean` and removed the root `node_modules`. Then reinstalled everything and built the whole thing over and over again. The problem with `useStores` won't go away... The stores cannot be injected in the `shared` package - it's `undefined`

**Comment by danieljwestman** [comment]
Update: if I manually remove the `vue` package from all `node_modules` except in the root it's all good again... 😅 But I have to do this every time I've updated 🤔

**Comment by joselcvarela** [comment]
Moved this into discussion since this does not look like a Directus issue

**Comment by joselcvarela** [comment]
Hello @danieljwestman
Since you have done `lerna clean`, do you have the `node_modules` folders again inside packages?
